darts/tests/models/forecasting/test_residuals.py

```diff
@@ -612,3 +619,304 @@ def test_sample_weight(self, config):
+            )
+        )
+        # historical forecasts
+        vals = np.random.random((10, 1, 100))
+        if not stochastic_pred:
+            vals = np.quantile(vals, q, axis=2).transpose(1, 0, 2)
+            comp_names = pd.Index(
+                likelihood_component_names(
+                    components=y.components,
+                    parameter_names=quantile_names(q=q_sorted),
+                )
+            )
+        else:
+            comp_names = y.components
+        vals = np.concatenate([vals, vals + 1], axis=1)
+        hfc = TimeSeries.from_times_and_values(
+            times=generate_index(start=y.start_time() + 10 * y.freq, length=10),
+            values=vals,
+            columns=comp_names,
+        )
+
+        y = [y, y]
+        if lpo:
+            hfc = [hfc, hfc]
+        else:
+            hfc = [[hfc, hfc], [hfc]]
+
+        metric_kwargs = {"component_reduction": None, "q": q}
+
+        model = NaiveDrift()
+
+        # return TimeSeries
+        bts = model.residuals(
+            series=y,
+            historical_forecasts=hfc,
+            metric=metric,
+            last_points_only=lpo,
+            metric_kwargs=metric_kwargs,
+        )
+        assert isinstance(bts, list) and len(bts) == 2
+        if lpo:
+            bts = [[bt] for bt in bts]
+
+        # `ae` with time and component reduction is equal to `mae` with component reduction
+        hfc_single = hfc[0][0] if not lpo else hfc[0]
+        bt_expected = metrics.ae(y[0], hfc_single, **metric_kwargs)
+        shape_expected = (len(hfc_single), len(q) * y[0].n_components)
+        for bt_list in bts:
+            for bt in bt_list:
+                assert bt.shape[:2] == shape_expected
+                assert bt.components.equals(q_comp_names_expected)
+                np.testing.assert_array_almost_equal(bt.values(), bt_expected)
+
+        # values only
+        bts = model.residuals(
+            series=y,
+            historical_forecasts=hfc,
+            metric=metric,
+            last_points_only=lpo,
+            metric_kwargs=metric_kwargs,
+            values_only=True,
+        )
+        assert isinstance(bts, list) and len(bts) == 2
+        if lpo:
+            bts = [[bt] for bt in bts]
+
+        # `ae` with time and component reduction is equal to `mae` with component reduction
+        for bt_list in bts:
+            for bt in bt_list:
+                assert bt.shape[:2] == shape_expected
+                np.testing.assert_array_almost_equal(bt[:, :, 0], bt_expected)
+
+    @pytest.mark.parametrize(
+        "config",
+        itertools.product(
+            [True, False],  # last_points_only
+            [False, True],  # from stochastic predictions (or predicted quantiles)
+        ),
+    )
+    def test_metric_quantiles_lpo_false(self, config):
+        """Tests backtest with different metric_kwargs based on historical forecasts generated on a sequence
+        `series` with last_points_only=False"""
```

Comment:
Same comment as above, contradiction between the values taken by `lpo` and the docstring/name of the test.